src/google/cloud/happybase/table.py

```diff
@@ -385,23 +387,16 @@ def scan(self, row_start=None, row_stop=None, row_prefix=None,
         row_start, row_stop, filter_chain = _scan_filter_helper(
             row_start, row_stop, row_prefix, columns, timestamp, limit, kwargs)
 
-        partial_rows_data = self._low_level_table.read_rows(
+        partial_rows_generator = self._low_level_table.yield_rows(
             start_key=row_start, end_key=row_stop,
             limit=limit, filter_=filter_chain)
 
-        # Mutable copy of data.
-        rows_dict = partial_rows_data.rows
-        while True:
-            try:
-                partial_rows_data.consume_next()
-                for row_key in sorted(rows_dict):
-                    curr_row_data = rows_dict.pop(row_key)
-                    # NOTE: We expect len(rows_dict) == 0, but don't check it.
-                    curr_row_dict = _partial_row_to_dict(
-                        curr_row_data, include_timestamp=include_timestamp)
-                    yield (row_key, curr_row_dict)
-            except StopIteration:
-                break
+        for partial_rowdata in partial_rows_generator:
+            curr_row_data = partial_rowdata
+            # NOTE: We expect len(rows_dict) == 0, but don't check it.
+            curr_row_dict = _partial_row_to_dict(
+                curr_row_data, include_timestamp=include_timestamp)
+            yield (curr_row_data.row_key, curr_row_dict)
```

Comment:
Is breaking the sort order worth the laziness here?  E.g., you have to work to restore it in the caller in the unit test.